driver/pace/driver/run.py

```diff
@@ -459,47 +477,65 @@ def __init__(
             comm: communication object behaving like mpi4py.Comm
         """
         self.config = config
-        communicator = pace.util.CubedSphereCommunicator.from_layout(
-            comm=comm, layout=self.config.layout
-        )
-        quantity_factory, stencil_factory = _setup_factories(
-            config=config, communicator=communicator
-        )
+        self.comm = comm
+        if self.config.performance_config.timer:
+            self.timestep_timer = pace.util.Timer()
+            self.timer = pace.util.Timer()
+        else:
+            self.timestep_timer = pace.util.NullTimer()
+            self.timer = pace.util.NullTimer()
```

Comment:
```suggestion
        if self.config.performance_config.timer:
            self.timestep_timer = pace.util.Timer()
            self.timer = pace.util.Timer()
        else:
            self.timestep_timer = pace.util.NullTimer()
            self.timer = pace.util.NullTimer()
```
You could move this concern out of driver and onto PerformanceConfig by making a property method `timer` on `PerformanceConfig` which gives a NullTimer in the right circumstances. Similarly for writing the results to disk.

Reply:
Great idea! This is now moved to `PerformanceConfig`.